oci/private/pull.bzl

```diff
@@ -479,78 +487,57 @@ oci_alias = repository_rule(
     },
 )
 
-_latest_build = """\
-load("@aspect_bazel_lib//lib:jq.bzl", "jq")
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
-
-jq(
-    name = "platforms",
-    srcs = ["manifest_list.json"],
-    filter = "[(.manifests // [])[] | .platform | .os + \\"/\\" + .architecture]",
-    # Print without newlines because it's too hard to indent that to fit under the generated
-    # starlark code below.
-    args = ["--compact-output"],
-)
-
-jq(
-    name = "mediaType",
-    srcs = ["manifest_list.json"],
-    filter = ".mediaType",
-    args = ["--raw-output"],
-)
-
+_LATEST_BUILD = """\
 sh_binary(
     name = "pin",
     srcs = ["pin.sh"],
-    data = [
-        ":mediaType",
-        ":platforms",
-        "@bazel_tools//tools/bash/runfiles",
-    ],
 )
 """
 
-_pin_sh = """\
+_PIN_SH = """\
 #!/usr/bin/env bash
-{rlocation}
 
-mediaType="$(cat $(rlocation {name}/mediaType.json))"
 echo -e "Replace your '{reponame}' declaration with the following:\n"
 
 cat <<EOF
 oci_pull(
     name = "{reponame}",
     digest = "sha256:{digest}",
     image = "{registry}/{repository}",
+    {optional_platforms}
+)
 EOF
-
-[[ $mediaType == "{manifestListType}" ]] && cat <<EOF
-    # Listing of all platforms that were found in the image manifest.
-    # You may remove any that you don't use.
-    platforms = $(cat $(rlocation {name}/platforms.json)),
-EOF
-
-echo ")"
 """
 
 def _pin_tag_impl(rctx):
     """Download the tag and create a repository that can produce pinning instructions"""
     downloader = _create_downloader(rctx)
-    downloader.download_manifest(rctx.attr.identifier, "manifest_list.json")
+    manifest, _ = downloader.download_manifest(rctx.attr.identifier, "manifest_list.json")
     result = rctx.execute(["shasum", "-a", "256", "manifest_list.json"])
     if result.return_code:
         msg = "shasum failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (result.stdout, result.stderr)
         fail(msg)
-    rctx.file("pin.sh", _pin_sh.format(
+
+    optional_platforms = ""
+    if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+        # calculate platforms
+        platforms = []
+        for submanifest in manifest["manifests"]:
+            parts = [submanifest["platform"]["os"], submanifest["platform"]["architecture"]]
+            if "variant" in submanifest["platform"]:
```

Comment:
this will also make future iterations easier for us when we add support for `os.version` and `os.features` which are a pain to deal within shell scripts.